staff/asier/tasks-list_express/src/logic/tasksLogic.js

```diff
@@ -0,0 +1,92 @@
+const tasksData = require('../data/tasksData')
+
+function validateText(text) {
+    if (!text) throw Error('You need to add some text')
+}
+
+function validateUsername(username) {
+    if (!username) throw Error('You need to add an user')
+}
+
+
+const tasksLogic = {
+
+    /* /// create task /// */
+
+    create(text, username) {
+
+        validateText(text)
+        validateUsername(username)
+
+        tasksData.create(text, username)        
+    },
+
+    /* /// List every task /// */
+
+    list() {
+        return tasksData.list()
+
+    },
+
+    /* /// Mark task as done  ///  */
+
+    markDone(id) {       
+        tasksData.markDone(id)
+
+    },
+
+    /* /// Remove task from tasklist */
+
+    remove(id) {
+
+        tasksData.delete(id)         
+    },
+
+    /* /// List tasks already done /// */
+
+    listDone() {
+        return tasksData.listDone()
```

Comment:
taskLogic MUST do the logic contained inside this method from taskData, but not the taskData itself. responsibility is mixed here. please, pay attention to the description of the challenge in trello ,)